pkg/liqoctl/rest/nonce/create.go

```diff
@@ -81,6 +83,23 @@ func (o *Options) Create(ctx context.Context, options *rest.CreateOptions) *cobr
 
 func (o *Options) handleCreate(ctx context.Context) error {
 	opts := o.createOptions
+
+	if opts.OutputFormat != "" {
+		exitingTenantNs, err := o.namespaceManager.GetNamespace(ctx, o.clusterID.GetClusterID())
+		if err != nil && !kerrors.IsNotFound(err) {
+			opts.Printer.CheckErr(fmt.Errorf("unable to get tenant namespace: %v", output.PrettyErr(err)))
+			return err
+		}
+
+		nsname := tenantnamespace.GetNameForNamespace(o.clusterID.GetClusterID())
+		if exitingTenantNs != nil {
+			nsname = exitingTenantNs.Name
+		}
+		tenantNs := o.namespaceManager.GenNamespaceObj(o.clusterID.GetClusterID(), nsname)
+		opts.Printer.CheckErr(o.output(tenantNs))
+		return nil
```

Comment:
I think is better to not use the `GetNameForNamespace()` function since we plan to make it private as can have unintended effects. My suggestion would be to modify the interface of the `GenNamespaceObj()` to accept an optional value `name *string`: if `name` is not set, it uses internally the `GetNameForNamespace()` function to generate the name., if set it uses the provided name (in this case, the tenant ns name if already exists).

Small note, I would move this logic directly to the output() function.